modules/azure-firewall/vnet.tf

```
@@ -0,0 +1,21 @@
+resource "azurerm_network_ddos_protection_plan" "main" {
+  count               = "${length(var.firewall)}"
+  name                = "ddospplan-${lookup(element(var.firewall, count.index), "location")}"
```

resource names are normally suffixed
```suggestion
  name                = "${lookup(element(var.firewall, count.index), "location")}-ddospplan"
```